app/models/user.rb

```diff
@@ -0,0 +1,20 @@
+class User < ApplicationRecord
+  before_save :downcase_email
+  validates :name, presence: true,
+    length: {maximum: Rails.application.config_for(:settings).max_name}
+  validates :email, presence: true,
+    length: {maximum: Rails.application.config_for(:settings).max_email},
+    format: {with: Rails.application.config_for(:settings).email_regex},
+    uniqueness: true
+  validates :password, presence: true,
+    length: {maximum: Rails.application.config_for(:settings).min_pass},
+  if: :password
+
+  has_secure_password
+
+  private
+
+  def downcase_email
+    sefl.email.downcase!
```

Comment:
em test thử chưa
code này ko ổn khi em truyền vào mail ko cần downcase
code đúng nên là 
```suggestion
    email.downcase!
```